src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -224,9 +232,12 @@ private async Task RegisterDesignerAttributeAsync(Document document, string desi
                 var documentId = document.Id;
```

Comment:
i can't actually mark the line since github has collapsed it.  but the crash is 5 lines above this on line 219:

```c#
                _ = updateService.SetProjectItemDesignerTypeAsync(document.FilePath, designerAttributeArgument)
.ReportNonFatalErrorAsync().CompletesAsyncOperation(asyncToken);
```

Right now the above added code is a partial mitigation, but not a full fix.  It's still racey in that we can do the check, see that things are ok, then immediately call into the code that crashes.  

It may be acceptable to have a mitigated fix.  however, it's been our experience in practice that without a full fix, people will still run into this.  So i would still recommend having them either fix this, or havin us go back to the UI thread to verify that things are ok.

One approach that would potentially work would be to collect all the doc information for an entire project on the bg.  You could then marshal back to the UI thread, see if the project was there, then call this with all the data at once (preventing one marshal per doc to the UI thread).

Alternatively, if there's some sort of appropriate *other* sync context that can be used to determine both if the project is gone *and* you can report this, then we could use that sync context to avoid the UI thread.

Tagging @sharwell and @davkean  for thoughts here.